### PR TITLE
User timezone caching

### PR DESCRIPTION
**Pull request by KeithRaven**
Fix for issue #15536

### CODE REVIEW

**Comment by hishamco** [comment]
Fix the merge conflict

File: src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/TimeZoneUserEventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Users.Handlers;
+using OrchardCore.Users.TimeZone.Services;
+
+namespace OrchardCore.Users.TimeZone.Handlers;
+internal class TimeZoneUserEventHandler : UserEventHandlerBase
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/TimeZoneUserEventHandler.cs:12]:
```suggestion

public class TimeZoneUserEventHandler : UserEventHandlerBase
```

File: src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/TimeZoneUserEventHandler.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{
+    private readonly UserTimeZoneService _userTimeZoneService;
+
+    public TimeZoneUserEventHandler(UserTimeZoneService userTimeZoneService) => _userTimeZoneService = userTimeZoneService;
+
+    public override async Task DeletedAsync(UserDeleteContext context)
+    {
+        await _userTimeZoneService.UpdateUserTimeZoneAsync(context.User);
+    }
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/TimeZoneUserEventHandler.cs:21]:
```suggestion
    public override async Task DeletedAsync(UserDeleteContext context)
        => await _userTimeZoneService.UpdateUserTimeZoneAsync(context.User);
```

File: src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/TimeZoneUserEventHandler.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    public override async Task UpdatedAsync(UserUpdateContext context)
+    {
+        await _userTimeZoneService.UpdateUserTimeZoneAsync(context.User);
+    }
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/TimeZoneUserEventHandler.cs:26]:
```suggestion
    public override async Task UpdatedAsync(UserUpdateContext context)
        => await _userTimeZoneService.UpdateUserTimeZoneAsync(context.User);
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Comment by KeithRaven** [comment]
Switched to draft as there's still a point for discussion in the linked issue.

File: src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/TimeZoneUserEventHandler.cs (hunk 1 above)
Comment (KeithRaven) [reply src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/TimeZoneUserEventHandler.cs]: Thanks updated

**Comment by hishamco** [comment]
@KeithRaven is this something you will revise soon?

**Comment by MikeAlhayek** [comment]
for instructions on how to resolve the merge conflicts due to #16572 please follow the step listed in [this comment](https://github.com/OrchardCMS/OrchardCore/issues/16572#issuecomment-2291836365).

**Comment by hishamco** [comment]
@deanmarcussen can we reopen this and accept this PR?

**Comment by KeithRaven** [comment]
Apologies for letting this slide.  If I can get an opinion on the question I asked in the issue re appropriate absolute expiration time https://github.com/OrchardCMS/OrchardCore/issues/15536#issuecomment-2080427836 then I'll get this tidied up.

**Comment by deanmarcussen** [comment]
conceptually fine, but code has since been refactored by someone else and the cache inside the `UserTimeZoneService` is now essentially useless, so the new issue is what really needs fixing (at which point this could be visited as the idea is good, but where it is, is no longer useful.)

refer : [#16392](https://github.com/OrchardCMS/OrchardCore/issues/16932)

**Pull request closed without merging**